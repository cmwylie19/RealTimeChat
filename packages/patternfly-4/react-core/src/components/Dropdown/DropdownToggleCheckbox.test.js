"use strict";
exports.__esModule = true;
var React = require("react");
var enzyme_1 = require("enzyme");
var DropdownToggleCheckbox_1 = require("./DropdownToggleCheckbox");
var props = {
    onChange: jest.fn(),
    isChecked: false
};
test('controlled', function () {
    var view = enzyme_1.shallow(<DropdownToggleCheckbox_1.DropdownToggleCheckbox isChecked id="check" aria-label="check"/>);
    expect(view).toMatchSnapshot();
});
test('uncontrolled', function () {
    var view = enzyme_1.shallow(<DropdownToggleCheckbox_1.DropdownToggleCheckbox id="check" aria-label="check"/>);
    expect(view).toMatchSnapshot();
});
test('with text', function () {
    var view = enzyme_1.shallow(<DropdownToggleCheckbox_1.DropdownToggleCheckbox id="check" isDisabled aria-label="check">Some text</DropdownToggleCheckbox_1.DropdownToggleCheckbox>);
    expect(view).toMatchSnapshot();
});
test('isDisabled', function () {
    var view = enzyme_1.shallow(<DropdownToggleCheckbox_1.DropdownToggleCheckbox id="check" isDisabled aria-label="check"/>);
    expect(view).toMatchSnapshot();
});
test('3rd state', function () {
    var view = enzyme_1.shallow(<DropdownToggleCheckbox_1.DropdownToggleCheckbox id="check" isChecked={null} aria-label="check"/>);
    expect(view).toMatchSnapshot();
});
test('passing class', function () {
    var view = enzyme_1.shallow(<DropdownToggleCheckbox_1.DropdownToggleCheckbox label="label" className="class-123" id="check" isChecked aria-label="check"/>);
    expect(view).toMatchSnapshot();
});
test('passing HTML attribute', function () {
    var view = enzyme_1.shallow(<DropdownToggleCheckbox_1.DropdownToggleCheckbox label="label" aria-labelledby="labelId" id="check" isChecked aria-label="check"/>);
    expect(view).toMatchSnapshot();
});
test('checkbox passes value and event to onChange handler', function () {
    var newValue = true;
    var event = {
        target: { checked: newValue }
    };
    var view = enzyme_1.mount(<DropdownToggleCheckbox_1.DropdownToggleCheckbox id="check" {...props} aria-label="check"/>);
    view.find('input').simulate('change', event);
    expect(props.onChange.mock.calls[0][0]).toBe(newValue);
    expect(props.onChange.mock.calls[0][1]).toMatchObject(event);
});
