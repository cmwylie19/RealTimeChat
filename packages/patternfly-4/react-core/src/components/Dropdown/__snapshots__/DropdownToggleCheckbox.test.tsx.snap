// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`3rd state 1`] = `
<label
  className="pf-c-dropdown__toggle-check"
  htmlFor="check"
>
  <Checkbox
    aria-invalid={false}
    aria-label="check"
    className=""
    id="check"
    isChecked={null}
    isDisabled={false}
    isValid={true}
    onChange={[Function]}
  />
</label>
`;

exports[`controlled 1`] = `
<label
  className="pf-c-dropdown__toggle-check"
  htmlFor="check"
>
  <Checkbox
    aria-invalid={false}
    aria-label="check"
    className=""
    id="check"
    isChecked={true}
    isDisabled={false}
    isValid={true}
    onChange={[Function]}
  />
</label>
`;

exports[`isDisabled 1`] = `
<label
  className="pf-c-dropdown__toggle-check"
  htmlFor="check"
>
  <Checkbox
    aria-invalid={false}
    aria-label="check"
    className=""
    id="check"
    isChecked={false}
    isDisabled={true}
    isValid={true}
    onChange={[Function]}
  />
</label>
`;

exports[`passing HTML attribute 1`] = `
<label
  className="pf-c-dropdown__toggle-check"
  htmlFor="check"
>
  <Checkbox
    aria-invalid={false}
    aria-label="check"
    aria-labelledby="labelId"
    className=""
    id="check"
    isChecked={true}
    isDisabled={false}
    isValid={true}
    label="label"
    onChange={[Function]}
  />
</label>
`;

exports[`passing class 1`] = `
<label
  className="pf-c-dropdown__toggle-check class-123"
  htmlFor="check"
>
  <Checkbox
    aria-invalid={false}
    aria-label="check"
    className=""
    id="check"
    isChecked={true}
    isDisabled={false}
    isValid={true}
    label="label"
    onChange={[Function]}
  />
</label>
`;

exports[`uncontrolled 1`] = `
<label
  className="pf-c-dropdown__toggle-check"
  htmlFor="check"
>
  <Checkbox
    aria-invalid={false}
    aria-label="check"
    className=""
    id="check"
    isChecked={false}
    isDisabled={false}
    isValid={true}
    onChange={[Function]}
  />
</label>
`;

exports[`with text 1`] = `
<label
  className="pf-c-dropdown__toggle-check"
  htmlFor="check"
>
  <Checkbox
    aria-invalid={false}
    aria-label="check"
    className=""
    id="check"
    isChecked={false}
    isDisabled={true}
    isValid={true}
    onChange={[Function]}
  />
  <span
    aria-hidden="true"
    className="pf-c-dropdown__toggle-text"
    id="check-text"
  >
    Some text
  </span>
</label>
`;
